File: htdocs/website/order.php
<?php
    $area = filter_var(trim($_POST['ServiceArea']), FILTER_SANITIZE_SPECIAL_CHARS);
    $address = filter_var(trim($_POST['address']), FILTER_SANITIZE_SPECIAL_CHARS);
    $city = filter_var(trim($_POST['city']), FILTER_SANITIZE_SPECIAL_CHARS);
    $country = filter_var(trim($_POST['country']), FILTER_SANITIZE_SPECIAL_CHARS);
    $date = filter_var(trim($_POST['date']), FILTER_SANITIZE_SPECIAL_CHARS);
    $task = filter_var(trim($_POST['taskDescription']), FILTER_SANITIZE_SPECIAL_CHARS);
    $additional = filter_var(trim($_POST['details']), FILTER_SANITIZE_SPECIAL_CHARS);
    
    
    /*if(mb_strlen($login) < 5 || mb_strlen($login) > 90){
        echo"Login must be from 5 to 90 characters";
        exit();
    }else if(mb_strlen($name) < 3 || mb_strlen($name) > 50){
        echo"Name must be from 3 to 50 characters";
        exit();
    }else if(mb_strlen($password) < 2 || mb_strlen($password) > 6){
        echo"Password must be from 2 to 6 characters";
        exit();
    }

    setcookie('user', $login, time() + 3600, "/"); // Устанавливаем cookie на час
    header('Location: MainSite.php'); // Перенаправление на MainSite.html
    exit(); */

    //$password = md5($password."qweqweqwe123");

    if (empty($area) || empty($address) || empty($city)) {
        die("Ошибка: все поля должны быть заполнены!");
    }
    
    $mysql = new mysqli('localhost', 'root', '', 'balti24db');
    if ($mysql->connect_error) {
        die("Ошибка подключения: " . $mysql->connect_error);
    }
    
    $query = "INSERT INTO tasks (area, address, city, country, date, task, additional) 
              VALUES ('$area', '$address', '$city', '$country', '$date', '$task', '$additional')";
    
    if (!$mysql->query($query)) {
        die("Ошибка запроса: " . $mysql->error);
    }
    
    $mysql->close();
    header('Location: Stask.html');
    ?>

File: htdocs/website/order.c
<?php
session_start(); // Начинаем сессию

if (!isset($_SESSION['user_id'])) {
    die("Ошибка: пользователь не авторизован!");
}

// Получаем данные из формы
$area = filter_var(trim($_POST['ServiceArea'] ?? ''), FILTER_SANITIZE_SPECIAL_CHARS);
$address = filter_var(trim($_POST['address'] ?? ''), FILTER_SANITIZE_SPECIAL_CHARS);
$city = filter_var(trim($_POST['city'] ?? ''), FILTER_SANITIZE_SPECIAL_CHARS);
$country = filter_var(trim($_POST['country'] ?? ''), FILTER_SANITIZE_SPECIAL_CHARS);
$date = filter_var(trim($_POST['date'] ?? ''), FILTER_SANITIZE_SPECIAL_CHARS);
$task = filter_var(trim($_POST['taskDescription'] ?? ''), FILTER_SANITIZE_SPECIAL_CHARS);
$additional = filter_var(trim($_POST['details'] ?? ''), FILTER_SANITIZE_SPECIAL_CHARS);

// Проверка обязательных полей
if (empty($area) || empty($address) || empty($city)) {
    die("Ошибка: все поля должны быть заполнены!");
}

// Подключаемся к базе данных
$mysql = new mysqli('localhost', 'root', '', 'balti24db');
if ($mysql->connect_error) {
    die("Ошибка подключения к базе данных: " . $mysql->connect_error);
}

// Получаем логин пользователя из сессии
$user_id = $_SESSION['user_id'];
$result = $mysql->query("SELECT login FROM users WHERE id='$user_id'");

if ($result && $row = $result->fetch_assoc()) {
    $login = $row['login'];
} else {
    die("Ошибка: не удалось получить логин пользователя!");
}

// Вставляем данные в таблицу tasks
$query = "INSERT INTO tasks (login, area, address, city, country, date, task, additional)
          VALUES ('$login', '$area', '$address', '$city', '$country', '$date', '$task', '$additional')";

if (!$mysql->query($query)) {
    die("Ошибка запроса: " . $mysql->error);
}

// Закрываем соединение с базой данных и перенаправляем на страницу заказов
$mysql->close();
header('Location: STask.html');
exit();
?>
